src/proxy.ts

```diff
@@ -0,0 +1,109 @@
+import { fork, Sequence } from 'effection';
+import { AddressInfo } from 'net';
+import * as proxy from 'http-proxy';
+import * as http from 'http';
+import { listen, ReadyCallback } from './http';
+import { EventEmitter, forkOnEvent, resumeOnEvent } from './util';
+import * as trumpet from 'trumpet';
+import * as zlib from 'zlib';
+
+interface ProxyOptions {
+  port: number,
+  targetPort: number
+  inject?: string
+};
+
+function pipe(from, to) {
+  fork(function*() {
+    let writer = forkOnEvent(from, 'data', function*(chunk) {
+      to.write(chunk);
+    });
+
+    try {
+      yield resumeOnEvent(from, 'end');
+      to.end();
+    } finally {
+      writer.halt();
+    }
+  });
```

Comment:
I *think* that this is correct and doesn't leak anything. But a second set of eyes on this would be good.